### PR TITLE
[NATS] Add route data stream dimensions

### PR DESCRIPTION
**Pull request by ghost**
## What does this PR do?

Adds `route` data stream dimensions.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Relates #7745

### CODE REVIEW

File: packages/nats/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.3.5"
+  changes:
+    - description: Add route data stream dimensions.
```
Comment (gizas) [inline packages/nats/changelog.yml:4]:
```suggestion
    - description: Add route data stream dimensions. ECS fields added as part of this update
```

**Comment by gpop63** [comment]
<details>
<summary>Rollover test</summary>

```json
{
  "acknowledged": true,
  "shards_acknowledged": true,
  "old_index": ".ds-metrics-nats.route-default-2023.09.21-000001",
  "new_index": ".ds-metrics-nats.route-default-2023.09.21-000002",
  "rolled_over": true,
  "dry_run": false,
  "conditions": {}
}
```

</details> 

<details>
<summary>Mappings</summary>

```json
"nats": {
          "properties": {
            "route": {
              "properties": {
                "in": {
                  "properties": {
                    "bytes": {
                      "type": "long",
                      "time_series_metric": "counter"
                    },
                    "messages": {
                      "type": "long",
                      "time_series_metric": "counter"
                    }
                  }
                },
                "ip": {
                  "type": "ip",
                  "time_series_dimension": true
                },
                "out": {
                  "properties": {
                    "bytes": {
                      "type": "long",
                      "time_series_metric": "counter"
                    },
                    "messages": {
                      "type": "long",
                      "time_series_metric": "counter"
                    }
                  }
                },
                "pending_size": {
                  "type": "long",
                  "time_series_metric": "gauge"
                },
                "port": {
                  "type": "long",
                  "time_series_dimension": true
                },
                "remote_id": {
                  "type": "keyword",
                  "time_series_dimension": true
                },
                "subscriptions": {
                  "type": "long",
                  "time_series_metric": "gauge"
                }
              }
            },
```

</details>

File: packages/nats/data_stream/route/fields/ecs.yml [hunk 2]
```diff
@@ -2,5 +2,15 @@
...
   name: service.type
+- external: ecs
+  name: agent.id
+  dimension: true
+- external: ecs
+  name: container.id
+  dimension: true
+- external: ecs
+  name: host.name
+  dimension: true
```
Comment (agithomas) [inline packages/nats/data_stream/route/fields/ecs.yml:16]: NATS can be deployed nearly anywhere; on bare metal, in a VM, as a container, inside K8S, on a device, or whichever environment you choose.

Due to this, i would recommend having all common dimension fields included in the list. Please have a look at the TSDB migration guidelines document.

**Comment by agithomas** [comment]
Please have all the checklist verified and checked.

File: packages/nats/data_stream/route/fields/fields.yml [hunk 3]
```diff
@@ -8,6 +8,7 @@
         The number of subscriptions in this connection
     - name: remote_id
       type: keyword
+      dimension: true
```
Comment (agithomas) [inline packages/nats/data_stream/route/fields/fields.yml:11]: For all those fields which are not obvious to become as dimensions, it is recommended to add the reason to be added as dimension. Do you want to consider adding the reason for the three fields which are part of the fields.yml?

**Comment by constanca-m** [comment]
Hey @gpop63, can you mention if these changes on dimensions passed the testing kit?

File: packages/nats/data_stream/route/fields/fields.yml (hunk 3 above)
Comment (gpop63) [reply packages/nats/data_stream/route/fields/fields.yml:11]: Actually, adding `ip` and `port` as dimensions might not be required, what do you think?

As for reasons:
- `remote_id`: This field is marked as a dimension because it helps in identifying the remote connection to which the route is connected.

```json
{
  "server_id": "IjI0eDbuloh1mfhyKIRn3Y",
  "now": "2023-10-02T19:25:06.860945516Z",
  "num_routes": 2,
  "routes": [
    {
      "rid": 1,
      "remote_id": "Y7IUAUD6XiYJ7It4s1JeA9",
      "did_solicit": true,
      "is_configured": true,
      "ip": "172.20.0.8",
      "port": 6222,
      "pending_size": 0,
      "in_msgs": 0,
      "out_msgs": 0,
      "in_bytes": 0,
      "out_bytes": 0,
      "subscriptions": 0
    },
    {
      "rid": 2,
      "remote_id": "U48tXHK9Xz708uDwsqQoB9",
      "did_solicit": true,
      "is_configured": true,
      "ip": "172.20.0.9",
      "port": 6222,
      "pending_size": 0,
      "in_msgs": 0,
      "out_msgs": 0,
      "in_bytes": 0,
      "out_bytes": 0,
      "subscriptions": 0
    }
  ]
}
```

**Comment by gpop63** [comment]
@constanca-m tested all data streams with the testing tool, no documents were lost.

**Review by constanca-m — APPROVED** [review]

File: packages/nats/data_stream/route/fields/fields.yml (hunk 3 above)
Comment (agithomas) [reply packages/nats/data_stream/route/fields/fields.yml:11]: If every remote connection gets a remote_id , as you mentioned, IP and port are not important. Please do the tests to assert this.

Comment (gpop63) [reply packages/nats/data_stream/route/fields/fields.yml:11]: @agithomas from my tests all route connections get a `remote_id`, not sure if there can exist some edge cases in which the field won't be present.

Comment (agithomas) [reply packages/nats/data_stream/route/fields/fields.yml:11]: In the absence of uncertainity, lets keep IP and port to avoid data loss, i recommend.

**Review by agithomas — APPROVED** [review]
LGTM!
